pkg/dsdk/connection.go

```diff
@@ -290,6 +290,31 @@ func (c *ApiConnection) do(ctxt context.Context, method, url string, ro *greq.Re
 	if err != nil {
 		Log().Errorf("Couldn't stringify data, %s", ro.JSON)
 	}
+        // Strip all CHAP credentails before printing to logs
+        // Decode (Unmarshal) the []byte into a AppInstance struct
+        ai := &AppInstance{}
+        err = json.Unmarshal(sdata, ai)
+        if err != nil {
+	        fmt.Println(err)
+        }
+        // Replace CHAP credentials with ***stripped***
+	auth_field := false
+        for _, si := range ai.StorageInstances {
+            if si.Auth != nil {
+		auth_field = true
+                si.Auth.InitiatorUserName = "***stripped***"
+                si.Auth.InitiatorPassword = "***stripped***"
+                si.Auth.TargetUserName = "***stripped***"
+                si.Auth.TargetPassword = "***stripped***"
+            }
+        }
+
+        // Re-encode (Marshal) the JSON data to []byte
+	if auth_field == true {
+		sdata, err = json.Marshal(ai)
+	}
+
+        Log().Debugf("REST call going with following payload, %s", string(sdata))
```

Comment:
remove debug.

Reply:
This debug prints the API that is sent to Datera and without this statement, It will be very hard to isolate whether the issue falls on Datera side or driver side. It's been there for a long time and has helped in troubleshooting API issues. I suggest we keep this. Any CHAP secrets are stripped out.